crates/core/action-pipeline/src/actions/run_task.rs

```diff
@@ -95,7 +95,23 @@ pub async fn run_task(
     }
 
     let attempts_result = if is_cache_enabled {
-        let context = context.read().await;
+        if let Some(mutex_name) = &task.options.mutex {
```

Comment:
This feels... messy? But honestly, not sure there's another way :P

Reply:
Yep, it is. I'll give it some extra thought. I don't promise anything, but I'd like it to be a bit cleaner, if possible.

Reply:
@milesj 

I reworked the execution flow to avoid having nested `if-else` control structures.

This way the control flow is "simpler", but it involves 2 extra allocations and 2 extra indirections (for `Arc::new` and `RwLock::new`) that are necessary to match the types for the cases when `is_cache_enabled` is `true` and `false`.

One possibility can be to go back to a "more branched" approach, but extracting the repeated logic into its own function. For now I'll leave it as it is so we can discuss it.